Exercise_2.java

```diff
@@ -9,44 +20,136 @@ static class StackNode {
         StackNode(int data) 
         { 
             //Constructor here 
+            this.data=data;
+            this.next=null;
+            
+           
+
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
         //Write your code here for the condition if stack is empty. 
+
+        // If top is null return true else false
+        if (root != null){
+            return false;
+
+        }else{
+            return true;
+        }
+
     } 
   
     public void push(int data) 
     { 
-        //Write code to push data to the stack. 

```

Comment:
As you're implementing stack (Using LinkedList), ALL Operations push/pop/peek should be doing in O(1) time. So add/delete element at head of list which will be resulted in stack.